### PR TITLE
Implement editing application info

### PR DESCRIPTION
**Pull request by Puncher1**
## Summary

This PR implements the ability to edit the application info. This also adds 2 parameters to `AppInfo` which are needed to not interfere with the `edit` method.

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/appinfo.py [hunk 1]
```diff
@@ -232,6 +248,142 @@ def flags(self) -> ApplicationFlags:
...
+                payload['cover_image'] = cover_image
+            else:
+                payload['cover_image'] = utils._bytes_to_base64_data(cover_image)
+
+        if interactions_endpoint_url is not MISSING:
+            payload['interactions_endpoint_url'] = interactions_endpoint_url
+
+        if tags is not MISSING:
+            payload['tags'] = tags
+        print(payload)
```
Comment (Rapptz) [inline discord/appinfo.py:383]:
```suggestion

```

File: discord/appinfo.py [hunk 2]
```diff
@@ -232,6 +248,142 @@ def flags(self) -> ApplicationFlags:
...
+    ) -> AppInfo:
+        r"""|coro|
+
+        Edits the application info.
+
+        .. versionadded:: 2.4
+
+        .. note::
+
+            All parameters can be ``None`` to remove the respective one.
```
Comment (Rapptz) [inline discord/appinfo.py:274]:
```suggestion
        Edits the application info.

        .. versionadded:: 2.4
```

It's better to say this in every parameter's documentation. I know it's repetitive but it's harder to miss, plus doesn't break when a new parameter is added that isn't optional.

**Review by Rapptz — CHANGES_REQUESTED** [review]

File: discord/appinfo.py (hunk 1 above)
Comment (Puncher1) [reply discord/appinfo.py:383]: Debug leftover...

**Review by Rapptz — APPROVED** [review]
Thanks.

**Pull request merged**